Sources/NIOWebSocket/NIOWebSocketClientUpgrader.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+
+/// A `HTTPClientProtocolUpgrader` that knows how to do the WebSocket upgrade dance.
+///
+/// This upgrader assumes that the `HTTPClientUpgradeHandler` will create and send the upgrade request. 
+/// This upgrader also assumes that the `HTTPClientUpgradeHandler` will appropriately mutate the
+/// pipeline to remove the HTTP `ChannelHandler`s.
+public final class NIOWebClientSocketUpgrader: NIOHTTPClientProtocolUpgrader {
+    
+    /// RFC 6455 specs this as the required entry in the Upgrade header.
+    public var supportedProtocol: String = "websocket"
```

Comment:
do we need those `var`s `public`?

Reply:
They are part of the conformance to NIOHTTPClientProtocolUpgrader which is in the HTTP1 framework. They can be constants though, so I changed them.